packages/app/src/entities/token/hooks/use-get-balance.ts

```diff
@@ -17,24 +17,35 @@ const defaultTokenRecord = { name: '', symbol: '', decimals: 0 }
 
 const encodeAddress = (account: string) => new SorobanClient.Address(account).toScVal()
 
-export const useGetBalance = (tokenAddress: TokenAddress, userAddress?: string) => {
-  const [balanceInfo, setBalanceInfo] = useState<SorobanTokenRecord | null>()
+export const useGetBalance = (tokenAddressArray: TokenAddress[], userAddress?: string) => {
+  const [balanceInfo, setBalanceInfo] = useState<SorobanTokenRecord[] | null>(null)
   const makeInvoke = useMakeInvoke()
-  const tokenCache =
-    useContextSelector(MarketContext, (state) => state.tokens?.[tokenAddress]) ?? defaultTokenRecord
+  const tokensCache = useContextSelector(MarketContext, (state) => state.tokens) // ?? defaultTokenRecord
 
-  useEffect(() => {
+  const getBalances = useCallback(async () => {
     if (!userAddress) {
       setBalanceInfo(null)
       return
     }
-
-    const invoke = makeInvoke(tokenAddress)
     const balanceTxParams = [encodeAddress(userAddress)]
-    invoke('balance', decodei128, balanceTxParams).then((balance) => {
-      setBalanceInfo({ balance, ...tokenCache })
-    })
-  }, [makeInvoke, tokenAddress, userAddress, tokenCache])
+
+    const balancesArray: SorobanTokenRecord[] = []
+
+    await Promise.all(
+      tokenAddressArray.map((tokenAddress) => {
+        const tokenCache = tokensCache?.[tokenAddress] ?? defaultTokenRecord
+        const invoke = makeInvoke(tokenAddress)
+        return invoke('balance', decodei128, balanceTxParams).then((balance) => {
+          balancesArray.push({ balance, ...tokenCache })
+        })
+      }),
+    )
+    setBalanceInfo(balancesArray)
+  }, [makeInvoke, tokenAddressArray, userAddress, tokensCache])
```

Comment:
We can rewrite that without naming arrays as arrays and without mixing await and then in one function:

```suggestion
    const balances: SorobanTokenRecord[] = (await Promise.all(
      tokenAddresses.map((tokenAddress) => {
        const tokenCache = tokensCache?.[tokenAddress] ?? defaultTokenRecord
        const invoke = makeInvoke(tokenAddress)
        return invoke('balance', decodei128, balanceTxParams)
      }),
    )).map((balance) => ({ balance , ...tokenCache }))
    setBalanceInfo(balances)
  }, [makeInvoke, tokenAddresses, userAddress, tokensCache])
```

What do you think?

Reply:
That's worked!